### PR TITLE
chore(deps): Upgrade eslint-config-sentry@1.67.0

### PR DESCRIPTION
**Pull request by matejminar**
This includes the addition of eslint-plugin-testing-library (https://github.com/getsentry/eslint-config-sentry/pull/115).

### CODE REVIEW

File: tests/js/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  extends: ['plugin:testing-library/react'],
+  rules: {
+    'testing-library/render-result-naming-convention': 'off',
+    'testing-library/prefer-screen-queries': 'off',
+  },
```
Comment (billyvg) [inline tests/js/.eslintrc.js:6]: Should these be in `eslint-config-sentry`?

Comment (matejminar) [reply tests/js/.eslintrc.js:6]: Unfortunately, they get overridden by this extend if they are there.

We'll be able to get rid of `testing-library/prefer-screen-queries` line soon, just one more big file to go.

Once we get the extends in overrides feature with the upgraded eslint, we'll be able to delete this whole file.

Comment (matejminar) [reply tests/js/.eslintrc.js]: Actually, I can put them there 👍 Just had the wrong order when I was testing it previously.

**Review by evanpurkhiser — APPROVED** [review]

File: tests/js/spec/views/integrationPipeline/pipelineView.spec.jsx [hunk 2]
```diff
@@ -12,13 +12,13 @@ jest.mock(
...
 describe('PipelineView', () => {
-  it('renders awsLambdaProjectSelect', () => {
+  it('renders awsLambdaProjectSelect', async () => {
     mountWithTheme(
       <PipelineView pipelineName="awsLambdaProjectSelect" someField="someVal" />,
       {context: TestStubs.routerContext()}
     );
 
-    screen.findByText('mock_AwsLambdaProjectSelect');
+    await screen.findByText('mock_AwsLambdaProjectSelect');
```
Comment (evanpurkhiser) [inline tests/js/spec/views/integrationPipeline/pipelineView.spec.jsx:21]: Should it be getByText if this didn't need to resolve before?

Comment (matejminar) [reply tests/js/spec/views/integrationPipeline/pipelineView.spec.jsx:21]: @evanpurkhiser followed up here: https://github.com/getsentry/sentry/pull/29790

**Pull request merged**